envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/GroupChangeWatcher.kt

```diff
@@ -38,12 +38,12 @@ internal class GroupChangeWatcher(
         val groups = cache.groups()
         metrics.setCacheGroupsCount(groups.size)
         if (oldGroups != groups) {
-            emitNewGroupsEvent()
+            emitNewGroupsEvent(groups - oldGroups)
```

Comment:
Emitting events about removal of groups is not necessary, because we only need to create Snapshots for new groups. Java-control-plane takes care of removing unused Snapshots.